lib/kube/proxy/forwarder.go

```diff
@@ -2101,6 +2110,12 @@ func (f *Forwarder) getSPDYDialer(ctx authContext, sess *clusterSession, req *ht
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	proxier := utilnet.NewProxierWithNoProxyCIDR(http.ProxyFromEnvironment)
+	// When the target cluster is not served by this teleport service, the
+	// proxier must be nil to avoid using it through the reverse tunnel.
+	if sess.kubeAPICreds == nil {
+		proxier = nil
+	}
```

Comment:
nit: Move this to a `func (s *clusterSession) getProxier()` so we don't need to copy this edge case and comment everywhere we need to get the proxier.